src/plus/integrations/authentication/integrationAuthentication.ts

```diff
@@ -94,27 +100,58 @@ abstract class IntegrationAuthenticationProviderBase<ID extends IntegrationId =
 
 	protected abstract restoreSession(sessionId: string): Promise<ProviderAuthenticationSession | undefined>;
 
-	protected async deleteSecret(key: IntegrationAuthenticationKeys) {
+	protected async deleteSecret(key: IntegrationAuthenticationKeys, sessionId: string) {
 		await this.container.storage.deleteSecret(key);
+		await this.authenticationService.removeConfigured({
+			integrationId: this.authProviderId,
+			domain: isSelfHostedIntegrationId(this.authProviderId) ? sessionId : undefined,
+		});
 	}
 
 	protected async writeSecret(
 		key: IntegrationAuthenticationKeys,
 		session: ProviderAuthenticationSession | StoredSession,
 	) {
 		await this.container.storage.storeSecret(key, JSON.stringify(session));
+		await this.authenticationService.addConfigured({
+			integrationId: this.authProviderId,
+			domain: isSelfHostedIntegrationId(this.authProviderId) ? session.id : undefined,
```

Comment:
This scares me a since we could in the future change the session.id to be/mean something else and we'd implicitly break this.

I would suggest we add `domain` on to the session like we are doing with `cloud` to make it explicit.

Reply:
That's a good idea, though we would want to include a migration for existing sessions to include that like we did with `cloud`. @sergeibbb

Reply:
@eamodio I'm going to add a TODO here and write this up as post-release debt